### ISSUE TITLE
Error 500 when loading baseURL + query param without trailing slash

### ISSUE THREAD

**Issue by nicolaspayot** [nuxt/nuxt#21872]
### Environment

Nuxi 3.6.1

RootDir: /home/projects/nuxt-starter-pytnf9
Nuxt project info:

------------------------------
- Operating System: Linux
- Node Version:     v16.20.0
- Nuxt Version:     3.6.1
- Nitro Version:    2.5.2
- Package Manager:  npm@9.4.2
- Builder:          vite
- User Config:      app
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-pytnf9

- Go to `/s?q=something` and an error 500 should appear.
- Go to `/s/?q=something` and it works as expected.

### Describe the bug

When trying to load the page defined by a `baseURL` and with some query parameter **without trailing slash**, we get an error 500:

![image](https://github.com/nuxt/nuxt/assets/463319/6b6bb69b-5778-41c9-be7d-9ba28eaf09fe)

It works as expected if we try to load the page with a trailing slash after the `baseURL`:

![image](https://github.com/nuxt/nuxt/assets/463319/67295ec8-f206-4a04-8306-d5f5421071d8)

**Comment by danielroe** [comment]
Is this present when built or only in dev?

**Comment by nicolaspayot** [comment]
Indeed! A trailing slash is automatically added before the query parameter with production build.

**Comment by stephanelgrg** [comment]
Hello my dear Nico ❤️

This issue has been fixed, it's working with Nuxt 3.21.9, here is an example

<img width="1282" height="471" alt="Image" src="https://github.com/user-attachments/assets/22ce5bd5-50c2-4200-9c96-4b50dc4f5aee" />

https://stackblitz.com/edit/nuxt-starter-b1uzestb?file=app.vue

@danielroe 🎉

**Comment by danielroe** [comment]
nuxtathon closed @stephanelgrg
